VeganVirus/VeganVirus/DesktopAction.cpp

```diff
@@ -0,0 +1,93 @@
+#include "DesktopAction.h"
+#include <cmath>
+
+#define BOUND_DIFF 10
+
+DesktopAction::DesktopAction(Draw* draw, MouseManager& mouseManager, DesktopManager& desktopManager)
+	:Action(DESKTOP_ACTION_PROGRESS, L"meat.ico"), _mouseManager(mouseManager), _desktopManager(desktopManager)
+{
+	_draw = draw;
+	_mouseManager = mouseManager;
+	_desktopManager = desktopManager;
+	_actionTime = 0;
+}
+
+void DesktopAction::start()
+{
+	_actionTime = DESKTOP_ACTION_TIME;
+}
+
+
+void DesktopAction::update(double dt)
+{
+	if (_actionTime < 0)
+		return;
+	_actionTime -= dt;
+	Point_t mouseTemp = _mouseManager.getMousePosition();
+	POINT mousePos = { mouseTemp.x, mouseTemp.y };
+	std::vector<POINT> iconSpeedArr = iconsMoveFromIcons();
+	// loping for each icon and seting is new postion on the screen base on interaction between the icons and the curser
```

Comment:
Comment is too long. Also, there are multiple typos.
If you had to write a long comment (which isn't the case here), you can break it into multiple lines.

```suggestion
	// going over all icons, calculating their velocity and updating their position
```